test/blackbox/common/DDSBlackboxTestsDataReader.cpp

```diff
@@ -414,6 +414,80 @@ TEST(DDSDataReader, default_qos_large_history_depth)
     ASSERT_TRUE(reader.isInitialized());
 }
 
+/*
+* This test creates a DataReader using DATAREADER_QOS_USE_TOPIC_QOS and checks if DataReader QoS
+* are correctly set from Topic QoS. It also checks if one of the QoS that is not in common with Topic QoS
+* is correctly set as the default one.
+*/
+TEST(DDSDataReader, datareader_qos_use_topic_qos)
+{
+    using namespace eprosima::fastdds::dds;
+
+    // Set custom Topic QoS
+    TopicQos topic_qos;
+    DurabilityServiceQosPolicy durability_service;
+    durability_service.history_kind = eprosima::fastdds::dds::KEEP_ALL_HISTORY_QOS;
+    topic_qos.durability_service(durability_service);
+    ReliabilityQosPolicy reliability;
+    reliability.kind = eprosima::fastdds::dds::RELIABLE_RELIABILITY_QOS;
+    topic_qos.reliability(reliability);
+    DurabilityQosPolicy durability;
+    durability.kind = eprosima::fastdds::dds::TRANSIENT_LOCAL_DURABILITY_QOS;
+    topic_qos.durability(durability);
+    DeadlineQosPolicy deadline;
+    deadline.period = {0, 500000000};
+    topic_qos.deadline(deadline);
+    LatencyBudgetQosPolicy latency;
+    latency.duration = 0;
+    topic_qos.latency_budget(latency);
+    LivelinessQosPolicy liveliness;
+    liveliness.kind = eprosima::fastdds::dds::MANUAL_BY_PARTICIPANT_LIVELINESS_QOS;
+    topic_qos.liveliness(liveliness);
+    ResourceLimitsQosPolicy resource_limit;
+    resource_limit.max_samples = 1000;
+    topic_qos.resource_limits(resource_limit);
+    LifespanQosPolicy lifespan;
+    lifespan.duration = {5, 0};
+    topic_qos.lifespan(lifespan);
+    OwnershipQosPolicy ownership;
+    ownership.kind = eprosima::fastdds::dds::EXCLUSIVE_OWNERSHIP_QOS;
+    topic_qos.ownership(ownership);
+    DataRepresentationQosPolicy data_rep;
+    data_rep.m_value.push_back(DataRepresentationId_t::XCDR2_DATA_REPRESENTATION);
+    topic_qos.representation(data_rep);
```

Comment:
Good catch thank you! Even though, DestinationOrderQos is not supported yet for the creation of a topic so it will return an error, it can be checked in the unittest.